src/sagemaker/serve/builder/model_builder.py

```diff
@@ -616,6 +622,10 @@ def build(
                 )
                 if hf_model_md.get("pipeline_tag") == "text-generation":  # pylint: disable=R1705
                     return self._build_for_tgi()
+                elif self._can_fit_on_single_gpu():
+                    return self._build_for_transformers()
+                elif _determine_engine_for_model(self.model, None, None) == FasterTransformerModel:
```

Comment:
i think this is close. I was thinking smth like this though:
```
1. using the config.json, get model_task
2. is model_task in DEEPSPEED_RECOMMENDED_ARCHITECTURES or FASTER_TRANSFORMER_RECOMMENDED_ARCHITECTURES then build for DJL
3. if not, then build for transformers
```

Reply:
Addressed, thanks